### ISSUE TITLE
False positive: empty() usage suggests to use null-comparison when variable is not null

### ISSUE THREAD

**Issue by Azenet** [kalessil/phpinspectionsea#1053]
Hi,

| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-Positive                          |
| Plugin         | Php Inspections (EA Extended) 3.0.6.1 |
| Language level | PHP 7.2 |

Current behaviour
---
With the following code, the `You should probably use '$c === null' instead.` inspection is triggered on the `if(empty())` condition:
```php
<?php

if (true) {
    $c = file_get_contents('...');
} else {
    $c = null;
}

if (empty($c)) {
    // ...
}
```

Expected behaviour
---
The inspection should only trigger if the variable can only be null.

Environment details
---
PhpStorm 2018.2.2
Build #PS-182.4129.45, built on August 22, 2018
JRE: 1.8.0_152-release-1248-b8 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.18.9-arch1-1-ARCH

**Comment by kalessil** [comment]
Tested with newer version, couldn't reproduce. Would you please try updating to PhpStorm 2018.2.3?

**Comment by Azenet** [comment]
Sorry, I didn't notice PhpStorm was outdated, I can't reproduce after updating, closing.
